config/consensus.go

```diff
@@ -301,6 +301,10 @@ type ConsensusParams struct {
 	// provide greater isolation for clear state programs
 	IsolateClearState bool
 
+	// should inner appl txns be allowed to call v4 (and v5) applications?
+	// this also disallows application downgrades
+	AllowV4InnerAppls bool
```

Comment:
@jannotti I have a semi-formed thought that's easiest to explain by showing.  I opened https://github.com/jannotti/go-algorand/pull/8 to capture the thought.  Can you have a look?

I specifically opened the PR as a draft.  _If_ the idea is deemed worth retaining, I think there's additional effort (primarily, inline comment updates) needed prior to merging.

I'm _not_ strongly attached to the idea.  If the change is unwieldy or not a big enough improvement, I'll happily close it.

Reply:
Took the idea, but did it from scratch, as it was easier to force myself to think everything though that way.